bgpd/bgp_nht.c

```diff
@@ -1233,7 +1241,16 @@ void evaluate_paths(struct bgp_nexthop_cache *bnc)
 			SET_FLAG(path->flags, BGP_PATH_IGP_CHANGED);
 
 		path_valid = CHECK_FLAG(path->flags, BGP_PATH_VALID);
-		if (path_valid != bnc_is_valid_nexthop) {
+		/* redistribute routes are always valid
+		 * if nht is called with redistribute routes, the vpn
+		 * exportation needs to be triggered
+		 */
+		if ((path->sub_type == BGP_ROUTE_REDISTRIBUTE) &&
+		    bnc_is_valid_nexthop && safi == SAFI_UNICAST &&
+		    (bgp_path->inst_type != BGP_INSTANCE_TYPE_VIEW))
```

Comment:
`(path->sub_type == BGP_ROUTE_REDISTRIBUTE)` `(bgp_path->inst_type != BGP_INSTANCE_TYPE_VIEW)` do not need parenthesis

Reply:
if you say in comment that "redistribute routes are always valid", why do you test `bnc_is_valid_nexthop` ?

Reply:
`bgp_path->inst_type != BGP_INSTANCE_TYPE_VIEW` mandatory ?

Reply:
> `(path->sub_type == BGP_ROUTE_REDISTRIBUTE)` `(bgp_path->inst_type != BGP_INSTANCE_TYPE_VIEW)` do not need parenthesis

removed parenthesis

Reply:
> if you say in comment that "redistribute routes are always valid", why do you test `bnc_is_valid_nexthop` ?

true, otherwise routes would not be redistributed.
removing the check

Reply:
> `bgp_path->inst_type != BGP_INSTANCE_TYPE_VIEW` mandatory ?

yes. redistribute can happen to bgp view instances too.
but leaking can not.
so this test needs to be kept (note that similar code exists in bgp_redistribute_add() code).